scripts/validate_docstrings.py

```diff
@@ -505,42 +466,107 @@ def validate_one(func_name):
+    return {'type': doc.type,
+            'docstring': doc.clean_doc,
+            'deprecated': doc.deprecated,
+            'file': doc.source_file_name,
+            'file_line': doc.source_file_def_line,
+            'github_link': doc.github_url,
+            'errors': errs,
+            'warnings': wrns,
+            'examples_errors': examples_errs}
+
+
+def validate_all():
+    """
+    Execute the validation of all docstrings, and return a dict with the
+    results.
+
+    Returns
+    -------
+    dict
+        A dictionary with an item for every function/method... containing
+        all the validation information.
+    """
+    result = {}
+    seen = {}
+
+    # functions from the API docs
+    api_items = list(get_api_items())
+    for func_name, func_obj, section, subsection in api_items:
+        doc_info = validate_one(func_name)
+        result[func_name] = doc_info
 
-    if not errs:
-        sys.stderr.write('Docstring for "{}" correct. :)\n'.format(func_name))
+        shared_code_key = doc_info['file'], doc_info['file_line']
+        shared_code = seen.get(shared_code_key, '')
+        result[func_name].update({'in_api': True,
+                                  'section': section,
+                                  'subsection': subsection,
+                                  'shared_code_with': shared_code})
 
-    if examples_errs:
-        sys.stderr.write(_output_header('Doctests'))
-        sys.stderr.write(examples_errs)
+        seen[shared_code_key] = func_name
+
+    # functions from introspecting Series, DataFrame and Panel
+    api_item_names = set(list(zip(*api_items))[0])
+    for class_ in (pandas.Series, pandas.DataFrame, pandas.Panel):
+        for member in inspect.getmembers(class_):
+            func_name = 'pandas.{}.{}'.format(class_.__name__, member[0])
+            if (not member[0].startswith('_') and
+                    func_name not in api_item_names):
+                doc_info = validate_one(func_name)
+                result[func_name] = doc_info
+                result[func_name]['in_api'] = False
+
+    return result
 
-    return len(errs)
 
+def main(func_name, fd):
+    def header(title, width=80, char='#'):
+        full_line = char * width
+        side_len = (width - len(title) - 2) // 2
+        adj = '' if len(title) % 2 == 0 else ' '
+        title_line = '{side} {title}{adj} {side}'.format(side=char * side_len,
+                                                         title=title,
+                                                         adj=adj)
 
-def main(function):
-    if function is None:
-        return validate_all()
+        return '\n{full_line}\n{title_line}\n{full_line}\n\n'.format(
+            full_line=full_line, title_line=title_line)
+
+    if func_name is None:
+        json_doc = validate_all()
+        fd.write(json.dumps(json_doc))
     else:
-        return validate_one(function)
+        doc_info = validate_one(func_name)
```

Comment:
Is the reason why we write different output for one vs all simply console readability? Wondering if we shouldn't align the output regardless of how this is invoked (maybe a separate PR depending on complexity)

Reply:
I see your point, but I think the lack of consistency is necessary in this case.

If the script is called for one object (a user validating the docstring they are changing), the "pretty" output in the console is what makes sense to return.

If the script is called for all objects (we as maintainers analyzing the status), a JSON with the information to be analyzed with pandas makes more sense than any console formatting.

Does it make sense?